tests/test_fabricpath_global.rb

```diff
@@ -216,4 +216,24 @@ def test_transition_delay
     assert_equal(20, @global.transition_delay,
                  'transition_delay not set to 20')
   end
+
+  def test_ttl_multicast
+    @global = FabricpathGlobal.new('default')
+    # test default value
+    assert_equal(32, @global.ttl_multicast,
+                 'Default multicast ttl not set to 32')
+    @global.ttl_multicast = 16
+    assert_equal(16, @global.ttl_multicast,
+                 'multicast ttl not getting set to 16')
+  end
+
+  def test_ttl_unicast
+    @global = FabricpathGlobal.new('default')
+    # test default value
+    assert_equal(32, @global.ttl_unicast,
+                 'Default unicast ttl not set to 32')
+    @global.ttl_unicast = 40
+    assert_equal(40, @global.ttl_unicast,
+                 'unicast ttl not getting set to 40')
+  end
```

Comment:
OK I can re-work this.

Thx,
deepak

From: Chris Van Heuveln <notifications@github.com<mailto:notifications@github.com>>
Reply-To: cisco/cisco-network-node-utils <reply@reply.github.com<mailto:reply@reply.github.com>>
Date: Wednesday, November 11, 2015 at 7:59 AM
To: cisco/cisco-network-node-utils <cisco-network-node-utils@noreply.github.com<mailto:cisco-network-node-utils@noreply.github.com>>
Cc: Cisco Employee <dcherian@cisco.com<mailto:dcherian@cisco.com>>
Subject: Re: [cisco-network-node-utils] adding fabricpath topo and fixed feature_set as a class method (#75)


In tests/test_fabricpath_global.rb<https://github.com/cisco/cisco-network-node-utils/pull/75#discussion_r44548286>:

> +    assert_equal(32, @global.ttl_multicast,
> +                 'Default multicast ttl not set to 32')
> +    @global.ttl_multicast = 16
> +    assert_equal(16, @global.ttl_multicast,
> +                 'multicast ttl not getting set to 16')
> +  end
> +
> +  def test_ttl_unicast
> +    @global = FabricpathGlobal.new('default')
> +    # test default value
> +    assert_equal(32, @global.ttl_unicast,
> +                 'Default unicast ttl not set to 32')
> +    @global.ttl_unicast = 40
> +    assert_equal(40, @global.ttl_unicast,
> +                 'unicast ttl not getting set to 40')
> +  end


1) Don't hard-code the default value
2) You can remove the duplication with:

def test_ttl_unicast
  @global = FabricpathGlobal.new('default')
  %w(ttl_multicast, ttl_unicast).each do |prop|
    # test default value
    default = @global.send("default_#{prop}")
    @global.send("#{prop}=", default)
    assert_equal(default, @global.send(prop),
                 "#{prop} : Default not set to #{default}")
    # test non-default value
    non_default = default + 8
    @global.send("#{prop}=", non_default)
    assert_equal(non_default, @global.send(prop),
                 "#{prop} : Non-default not set to #{non_default}")
  end
end


-
Reply to this email directly or view it on GitHub<https://github.com/cisco/cisco-network-node-utils/pull/75/files#r44548286>.